src/storage/mod.rs

```diff
@@ -801,24 +803,26 @@ impl std::fmt::Debug for Storage {
     }
 }
 
-fn detect_mime(file_path: impl AsRef<Path>) -> &'static str {
+fn detect_mime(file_path: impl AsRef<Path>) -> Mime {
     let mime = mime_guess::from_path(file_path.as_ref())
```

Comment:
Can this `mime` creation be moved in the `_ =>` pattern instead of being created unconditionally?

Reply:
no, because the first `match` is actually on `Mime.as_str` and not on `file_path`

Reply:
Oh I see. Missed that part, sorry.